src/util/pointer.h

```diff
@@ -257,3 +257,12 @@ class SharedBuffer
 	unsigned int *refcount;
 };
 
+// This class is not thread-safe!
+class IntrusiveReferenceCounted {
+public:
+	virtual ~IntrusiveReferenceCounted() = default;
+	void grab() noexcept { ++m_refcount; }
+	void drop() noexcept { if (--m_refcount == 0) delete this; }
+private:
+	u64 m_refcount = 1;
```

Comment:
`unsigned int` definitely suffices here
(consider a 32 byte object, you'd need to fill 128GB with just instances of that object to overflow this)